contents/tutorials/monitor-aws-bedrock-calls.md

```diff
@@ -0,0 +1,354 @@
+
+```js file=src/app/api/generate-recipe/route.js
+// your existing imports
+
+import { PostHog } from 'posthog-node'; // import PostHog
+
+export async function POST(request) {
+  const posthog = new PostHog(
+    '<ph_project_api_key>',
+    {
+      host: '<ph_client_api_host>',
+    },
+  );
+
+  // rest of your code
+
+  try {
+    // existing code
+  } catch (error) {
+    // existing code
+  } finally {
+    // Call posthog.shutdown() to flush and send all pending events before the serverless function shuts down.
+    await posthog.shutdown();
+  }
+}
+```
+
+## 3. Capture events
+
+With our app set up, we can begin [capturing events](/docs/product-analytics/capture-events) with PostHog.
+
+### a. Successful requests
+To start, we capture a `bedrock_completion` event with properties related to the API request like:
+
+- `prompt`
+- `generation`
+- `prompt_token_count`
+- `generation_token_count`
+
+To do this, add a `posthog.capture()` call after receiving a response from Bedrock:
+
+```js file=src/app/api/generate-recipe/route.js
+// your existing code
+  try { 
+    // ... existing code
+
+    const command = new InvokeModelCommand(input);
+    const response = await client.send(command);
+    const rawRes = response.body;
+    const jsonString = new TextDecoder().decode(rawRes);
+    const parsedJSON = JSON.parse(jsonString);
+
+    // Add this
+    const { generation, prompt_token_count, generation_token_count} = parsedJSON;
+    posthog.capture({
+      distinctId: email,
+      event: 'bedrock_completion',
+      properties: {
+        prompt,
+        model_id: modelId,
+        generation: generation,
+        prompt_token_count: prompt_token_count,
+        generation_token_count: generation_token_count,
+      }
+    });
+
+    // rest of your code
+  }
+```
+
+Refresh your app and submit a few prompts. You should then see your events captured in the [PostHog activity tab](https://us.posthog.com/events).
+
+<ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/v1723020434/posthog.com/contents/Screenshot_2024-08-07_at_9.46.52_AM.png" 
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/v1723020434/posthog.com/contents/Screenshot_2024-08-07_at_9.47.04_AM.png" 
+  alt="AWS Bedrock events in PostHog" 
+  classes="rounded"
+/>
+
+### b. Costs
+
+To keep track of your generative AI costs, you can include additional properties to your event capture, namely:
+
+- `input_cost_in_dollars` i.e. `prompt_token_count` * `token_input_cost`
+- `output_cost_in_dollars` i.e. `generation_token_count` * `token_output_cost`
+- `total_cost_in_dollars` i.e. `input_cost_in_dollars + output_cost_in_dollars`
+
+You can view the token costs for your model in the [Bedrock pricing](https://aws.amazon.com/bedrock/pricing/) page. Since we're using **LLama 3.1 8B Instruct** in this tutorial, we set the `token_input_cost` and `token_output_cost` to the values for this model:
```

Comment:
```suggestion
You can view the token costs for your model in the [Bedrock pricing](https://aws.amazon.com/bedrock/pricing/) page. Since we're using **Llama 3.1 8B Instruct** in this tutorial, we set the `token_input_cost` and `token_output_cost` to the values for this model:
```